source/common/request_source_plugin_impl.cc

```diff
@@ -0,0 +1,106 @@
+#include "common/request_source_plugin_impl.h"
+
+#include "external/envoy/source/common/protobuf/message_validator_impl.h"
+#include "external/envoy/source/common/protobuf/utility.h"
+#include "external/envoy/source/exe/platform_impl.h"
+
+#include "api/client/options.pb.h"
+#include "api/request_source/request_source_plugin_impl.pb.h"
+
+#include "common/request_impl.h"
+#include "common/request_source_impl.h"
+
+namespace Nighthawk {
+
+std::string DummyRequestSourceConfigFactory::name() const {
+  return "nighthawk.dummy-request-source-plugin";
+}
+
+Envoy::ProtobufTypes::MessagePtr DummyRequestSourceConfigFactory::createEmptyConfigProto() {
+  return std::make_unique<nighthawk::request_source::DummyPluginRequestSourceConfig>();
+}
+
+RequestSourcePluginPtr
+DummyRequestSourceConfigFactory::createRequestSourcePlugin(const Envoy::Protobuf::Message& message,
+                                                           Envoy::Api::Api& api) {
+  const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+  nighthawk::request_source::DummyPluginRequestSourceConfig config;
+  Envoy::MessageUtil::unpackTo(any, config);
+  return std::make_unique<DummyRequestSourcePlugin>(config, api);
+}
+
+REGISTER_FACTORY(DummyRequestSourceConfigFactory, RequestSourcePluginConfigFactory);
+
+DummyRequestSourcePlugin::DummyRequestSourcePlugin(
+    const nighthawk::request_source::DummyPluginRequestSourceConfig& config, Envoy::Api::Api& api)
+    : RequestSourcePlugin{api}, dummy_value_{config.has_dummy_value()
+                                                 ? config.dummy_value().value()
+                                                 : std::numeric_limits<double>::infinity()} {}
+RequestGenerator DummyRequestSourcePlugin::get() {
+  RequestGenerator request_generator = []() {
+    Envoy::Http::RequestHeaderMapPtr header = Envoy::Http::RequestHeaderMapImpl::create();
+    auto returned_request_impl = std::make_unique<RequestImpl>(std::move(header));
+    return returned_request_impl;
+  };
+  return request_generator;
+}
+void DummyRequestSourcePlugin::initOnThread() {}
```

Comment:
this could perhaps fit nicely into a common base implementation as the default implementation, to save plugins that don't need it some bootstrapping.

Reply:
I'm not sure I entirely understand this comment.

Reply:
An example:

```c++

// Interface
class RequestSourcePlugin {
    virtual initOnThread() PURE;
};

// BaseImpl
class RequestSourcePluginBaseImpl {
    // Add default uninteresting behavior.
    void initOnThread() override {};
};

// Actual concrete implementations can derive from RequestSourcePluginBase above, e.g.
class RequestSourcePluginFooImpl : RequestSourcePluginBaseImpl {
    // no need to override initOnThread() here
};

```

Reply:
Understood.